extra_tests/snippets/syntax_non_utf8.py

```diff
@@ -0,0 +1,11 @@
+import os
+dir_path = os.path.dirname(os.path.realpath(__file__))
```

Comment:
```suggestion
import os

from testutils import assert_raises

dir_path = os.path.dirname(os.path.realpath(__file__))
```

Reply:
It's not working, becase cpython raise a `ValueError` but rustpython raise a `SyntaxError`, then the test will fail.

The snippet tests will run both on rustpython and cpython.

Reply:
I tried another fix by checking python platform implementation:

```python
import os
import platform

from testutils import assert_raises

dir_path = os.path.dirname(os.path.realpath(__file__))

# TODO: RUSTPYTHON, RustPython raises a SyntaxError here, but cpython raise a ValueError
error = SyntaxError if platform.python_implementation() == 'RustPython' else ValueError
with assert_raises(error):
    with open(os.path.join(dir_path , "non_utf8.txt")) as f:
        eval(f.read())

```